lib/reek/context_builder.rb

```diff
@@ -45,54 +50,88 @@ def initialize(syntax_tree)
     # RootContext -> children: 1 ModuleContext -> children: 1 MethodContext
     #
     # @return [Reek::Context::RootContext] tree of nested contexts
-    def process(exp)
+    def build(exp)
       context_processor = "process_#{exp.type}"
       if context_processor_exists?(context_processor)
         send(context_processor, exp)
       else
-        process_default exp
+        process exp
       end
       element
     end
 
+    # Handles every node for which we have no context_processor.
+    #
+    def process(exp)
+      exp.children.each do |child|
+        build(child) if child.is_a?(AST::Node)
+      end
+    end
```

Comment:
Somewhat tempted to suggest going all the way here:

```ruby
def process(exp)
  exp.children.grep(AST::Node).each(&method(:build))
end
```

…but I understand it might be considered too far. ;)